lib/storage/providers/redis.js

```diff
@@ -204,6 +204,51 @@ StorageRedis.prototype.getServiceOutagesSince = function (service, timestamp, ca
 };
 
 /**
+ * Get outage history for all services
+ * @param services
+ * @param timestamp
+ * @param maxItems
```

Comment:
I edited my comment above before reading yours.
What I suggested is using ``maxItemsPerService`` in the redis query and then slicing the final result to ``maxItems``. That should give you a decent granularity. How does it sound?